### PR TITLE
Issue #5950 - remove WebAppClassLoader logging during loadClass

### PR DESCRIPTION
**Pull request by joakime**
Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

**Comment by janbartel** [comment]
Such a pity this seems to be necessary: when things are really going wrong it was so helpful to turn on debug logging for the WebAppClassLoader and find out how classes were being loaded (or not).

**Comment by olamy** [comment]
@joakime  @janbartel if it's so helpful could we have something such `-Djetty.classloader.debug=true` and replace those LOG.debug with System.err/out? 
Or something such
```
private static final boolean DEBUG = System.getBoolean("jetty.classloader.debug")
then in code where we need it
if (DEBUG) 
   System.out/err
or (maybe more complicated)
  org.eclipse.jetty.util.log.Logger.getLogger(...).debug("")
```

File: jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClasspathPattern.java [hunk 1]
```diff
@@ -697,9 +693,8 @@ public boolean match(Class<?> clazz)
         {
             return combine(_packageOrNamePatterns, clazz.getName(), _locations, () -> TypeUtil.getLocationOfClass(clazz));
         }
-        catch (Exception e)
+        catch (Exception ignored)
         {
-            LOG.warn(e);
```
Comment (gregw) [inline jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClasspathPattern.java:702]: Probably should be e.printStackTrace() to replace warnings

File: jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClasspathPattern.java [hunk 2]
```diff
@@ -719,9 +714,8 @@ public boolean match(String name, URL url)
             {
                 return URIUtil.getJarSource(url.toURI());
             }
-            catch (URISyntaxException e)
+            catch (URISyntaxException ignored)
             {
-                LOG.ignore(e);
```
Comment (gregw) [inline jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClasspathPattern.java:724]: I wonder if we could get by with something like:
```java
if (LOG.isDebugEnabled())
   ignored.printStackTrace();
```

**Review by gregw — APPROVED** [review]
LGTM other than a few niggles.

File: jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClasspathPattern.java (hunk 2 above)
Comment (olamy) [reply jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClasspathPattern.java:724]: in order to fix the issue  we don't want to load the Logger class

**Comment by gregw** [comment]
Is it possible to test if `-verbose:class` is set, then we could do some stderr logging if it is?   But I'm still good with this and we can add in stderr logging if/when necessary

**Comment by joakime** [comment]
Sorry, this isn't ready.
There's still logging occurring during load class.

**Comment by joakime** [comment]
Sorry, the draft status was in error.
The extra logging I saw was not during loadClass.
Will merge shortly.

**Comment by jglick** [comment]
in https://github.com/eclipse/jetty.project/releases/tag/jetty-9.4.37.v20210219 FTR

**Pull request merged**